allennlp/tests/semparse/contexts/table_question_context_test.py

```diff
@@ -0,0 +1,103 @@
+# pylint: disable=no-self-use,invalid-name,protected-access,too-many-public-methods
+from allennlp.common.testing import AllenNlpTestCase
+from allennlp.semparse.contexts import TableQuestionContext
+from allennlp.data.tokenizers import WordTokenizer
+from allennlp.data.tokenizers.word_splitter import SpacyWordSplitter
+
+MAX_TOKENS_FOR_NUMBER = 6
+
+class TestTableQuestionContext(AllenNlpTestCase):
+    def setUp(self):
+        super().setUp()
+        self.tokenizer = WordTokenizer(SpacyWordSplitter(pos_tags=True))
+
+    def test_number_extraction(self):
+        question = """how many players on the 191617 illinois fighting illini men's basketball team
+                      had more than 100 points scored?"""
+        question_tokens = self.tokenizer.tokenize(question)
+        test_file = f'{self.FIXTURES_ROOT}/data/corenlp_processed_tables/TEST-7.table'
+        table_question_context = TableQuestionContext.read_from_file(test_file, question_tokens)
+        _, number_entities = table_question_context.get_entities_from_question()
+        assert number_entities == [("191617", "191617"), ("100", "100")]
+
+    def test_date_extraction(self):
+        question = "how many laps did matt kenset complete on february 26, 2006."
+        question_tokens = self.tokenizer.tokenize(question)
+        test_file = f'{self.FIXTURES_ROOT}/data/corenlp_processed_tables/TEST-8.table'
+        table_question_context = TableQuestionContext.read_from_file(test_file, question_tokens)
+        _, number_entities = table_question_context.get_entities_from_question()
+        assert number_entities == [("2", "february"), ("26", "26"), ("2006", "2006")]
+
+    def test_date_extraction_2(self):
+        question = """how many different players scored for the san jose earthquakes during their
+                      1979 home opener against the timbers?"""
+        question_tokens = self.tokenizer.tokenize(question)
+        test_file = f'{self.FIXTURES_ROOT}/data/corenlp_processed_tables/TEST-6.table'
+        table_question_context = TableQuestionContext.read_from_file(test_file, question_tokens)
+        _, number_entities = table_question_context.get_entities_from_question()
+        assert number_entities[0] == ("1979", "1979")
```

Comment:
Better to assert equality on the list - it's still possible with this test that `number_entities` has length > 1.  Same with another test below.

Reply:
Done.